### PR TITLE
accept a dictionary in plot colors

### LINKED ISSUE #8193: Passing a dictionary of columns names and colors
Related to this question:

http://stackoverflow.com/questions/25689558/pandas-bar-plot-specify-bar-color-by-column

Having the ability to natively pass in a dictionary of the column names and the desired color could be a nice addition to simplify specifying what color you want each line/bar to be.

### PR DESCRIPTION
**Pull request by Leostayner**
- [ ] closes #8193
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Line/bar plot accepts color by dictonary.
@elisamalzoni

### CODE REVIEW

File: pandas/tests/plotting/test_misc.py [hunk 1]
```diff
@@ -417,3 +417,23 @@ def test_get_standard_colors_no_appending(self):
         color_list = cm.gnuplot(np.linspace(0, 1, 16))
         p = df.A.plot.bar(figsize=(16, 7), color=color_list)
         assert p.patches[1].get_facecolor() == p.patches[17].get_facecolor()
+
+    @pytest.mark.slow
+    def test_dictionary_color(self):
+        # Test plot color dictionary format
```
Comment (gfyoung) [inline pandas/tests/plotting/test_misc.py]: Reference issue number as a comment

**Comment by jreback** [comment]
can you merge master

**Comment by Leostayner** [comment]
@jreback, Merged!

**Review by WillAyd — CHANGES_REQUESTED** [review]
Can you add a whatsnew note for v1.0.0 and update the docstring for `.plot` to reflect this?

**Comment by Leostayner** [comment]
Updated whatsnew. Now where should I update the `.plot` docstring? Here is where I think it should be, is it right?

https://github.com/pandas-dev/pandas/blob/c0050e071d0cc31501d4f730c27a81f3a126fc5c/pandas/plotting/_core.py#L855

https://github.com/pandas-dev/pandas/blob/c0050e071d0cc31501d4f730c27a81f3a126fc5c/pandas/plotting/_core.py#L924

**Comment by jreback** [comment]
can you merge master

**Comment by MarcoGorelli** [comment]
Hi @Leostayner - sorry to chase you up on this, just wanted to ask whether you're still working on it :)

**Comment by gfyoung** [comment]
@MarcoGorelli - given the silence, I think you're okay to finish this up if you want.

**Comment by MarcoGorelli** [comment]
Sure. Looks like it's gone stale, so I've taken it forward in #31071 (note to OP: you're welcome to reopen this if you want to keep working on it)

**Pull request closed without merging**